File: 10_week_2/fabulous_fibonacci.js
var fibonacci = [0, 1];

function fabulous_fibonacci(num) {
	if (num < 0) {
		return null;
	}

	while (fibonacci.length <= num) {
		var current = fibonacci[fibonacci.length-1];
		var previous = fibonacci[fibonacci.length-2];
		fibonacci.push(previous + current);
	}

	return fibonacci[num];
}

console.log(fabulous_fibonacci(9));

File: 10_week_2/fabulous_fibonacci.py
fibonacci = [0, 1]

def fabulous_fibonacci(num):
	if num < 0:
		return None

	while len(fibonacci) <= num:
		previous = fibonacci[len(fibonacci) - 2]
		current = fibonacci[len(fibonacci) - 1]

		fibonacci.append(previous + current)

	return fibonacci[num]

print(fabulous_fibonacci(9))
